arch/x86/boot/compressed/tpm/tpm2_cmds.c

```diff
@@ -84,11 +84,13 @@ int tpm2_extend_pcr(struct tpm *t, u32 pcr,
 	u16 size;
 	int ret = 0;
 
+	b = alloc_tpmbuff(t->intf, TPM_NO_LOCALITY);
```

Comment:
IMHO we should not be allocating the buffer in the middle of the extend, it should have already been allocated. Though we should be checking that the buffer was allocated, i.e. != NULL.

Reply:
If there are TPM fixes, lets bring them in before the AMD PR goes in. I think that goes for any other general fixes that are found that are not related to adding AMD support.

Reply:
Yes, this is an improvement to the existing tpm patches, so this should be captured in a delta commit.

Reply:
This function should probably be renamed, or split into two separate functions. The reason it is called here is that it also resets head/tail/len of that buffer.

Reply:
Rebased